gms/gossiper.cc

```diff
@@ -2275,14 +2289,19 @@ bool gossiper::is_alive(inet_address ep) const {
     if (ep == get_broadcast_address()) {
         return true;
     }
+    bool is_alive = _live_endpoints.contains(ep);
```

Comment:
`gossiper::is_alive` is on the hot paths, won't this regress performance?

Reply:
Still, it's two `unordered_set` searches instead of one. So assuming that the `unordered_set` search takes a significant portion of the operation (I don't know if that's the case, but we should consider it), we would increase that significant portion twice.

I'm quite worried about this change.

But wait. Do we actually need the `get_endpoint_state_for_endpoint_ptr` call below? Maybe we can just replace one search with another.

We should maintain an invariant: `_live_endpoints` is always a subset of `_endpoint_state_map` (which is used by `get_endpoint_state_for_endpoint_ptr`) (maybe the invariant is already satisfied? If not -- why?)

And then the second part of this function would be redundant, so we could have only a single search.

Reply:
The invariant should be satisfied, although I didn't audit the code to make sure it is.
FWIW, I've never seen the warning below...

We can do the check as a safety belt only in non-release builds so it won't hurt performance in production but would still be checked in testing.